### PR TITLE
Added e2e test config for aws-load-balancer-operator

### PR DESCRIPTION
**Pull request by thejasn**
**Description**
- Introduces a new custom workflow for subscribing to an optional operator `aws-load-balancer-operator`. This is required as the installation of this operator has a manual step which involves providing AWS credentials as a secret.
- Adds the required e2e test configs for the operator.

Signed-off-by: thejasn <thn@redhat.com>

### CODE REVIEW

File: ci-operator/step-registry/openshift/e2e/aws/aws-load-balancer/openshift-e2e-aws-aws-load-balancer-commands.sh [hunk 1]
```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+cat <<EOF | oc apply -f -
```
Comment (arjunrn) [inline ci-operator/step-registry/openshift/e2e/aws/aws-load-balancer/openshift-e2e-aws-aws-load-balancer-commands.sh:7]: I believe the repo is available when this script is run. So you should be able to create this from the repository directly. Better still read the relative path from an environment variable. Do not put the entire file in there. Adding more permissions later will be cumbersome.

File: ci-operator/config/openshift/aws-load-balancer-operator/openshift-aws-load-balancer-operator-main.yaml [hunk 2]
```diff
@@ -49,6 +64,31 @@ tests:
...
+    - ref: optional-operators-subscribe
+    test:
+    - as: test
+      cli: latest
+      commands: make test-e2e
+      from: src
+      resources:
+        requests:
+          cpu: 100m
+    workflow: optional-operators-ci-aws
```
Comment (arjunrn) [inline ci-operator/config/openshift/aws-load-balancer-operator/openshift-aws-load-balancer-operator-main.yaml:91]: Is the `workflow` field still required if the `steps` are specified.

File: ci-operator/config/openshift/insights-operator/openshift-insights-operator-release-4.10.yaml [hunk 3]
```diff
@@ -6,7 +6,7 @@ base_images:
   insights-operator-tests:
     name: insights-operator-tests
     namespace: ci
-    tag: latest
+    tag: "4.10"
```
Comment (arjunrn) [inline ci-operator/config/openshift/insights-operator/openshift-insights-operator-release-4.10.yaml]: This is unrelated to our change.

File: ci-operator/config/openshift/aws-load-balancer-operator/openshift-aws-load-balancer-operator-main.yaml [hunk 4]
```diff
@@ -20,6 +24,17 @@ images:
...
   to: aws-load-balancer-operator
+operator:
+  bundles:
+  - as: aws-load-balancer-operator-bundle
+    base_index: operator-index
+    dockerfile_path: bundle.Dockerfile
+  substitutions:
+  - pullspec: openshift.io/aws-load-balancer-operator:latest
+    with: pipeline:aws-load-balancer-operator
+promotion:
```
Comment (arjunrn) [inline ci-operator/config/openshift/aws-load-balancer-operator/openshift-aws-load-balancer-operator-main.yaml:35]: Do we need to promote the image?

File: ci-operator/step-registry/openshift/e2e/aws/aws-load-balancer/OWNERS [hunk 5]
```diff
@@ -0,0 +1 @@
+../OWNERS
```
Comment (arjunrn) [inline ci-operator/step-registry/openshift/e2e/aws/aws-load-balancer/OWNERS:1]: We should be the owners of this step and not just inherit it from the parent.

File: ci-operator/config/openshift/aws-load-balancer-operator/openshift-aws-load-balancer-operator-main.yaml [hunk 6]
```diff
@@ -49,6 +64,31 @@ tests:
     make test
   container:
     from: src
+- as: e2e-aws-operator
+  steps:
+    cluster_profile: aws
```
Comment (arjunrn) [inline ci-operator/config/openshift/aws-load-balancer-operator/openshift-aws-load-balancer-operator-main.yaml:69]: I would recommend that you extract these steps into a workflow in the step registry. But it's not a hard requirement for this PR.

**Review by arjunrn — COMMENTED** [review]
This is a good start. However I would encourage you to create a directory under the step directory https://github.com/openshift/release/blob/master/ci-operator/step-registry/OWNERS with the custom step we need. Refactor it so that it can take the relative path as parameter.

File: ci-operator/step-registry/openshift/e2e/aws/aws-load-balancer/openshift-e2e-aws-aws-load-balancer-commands.sh (hunk 1 above)
Comment (arjunrn) [reply ci-operator/step-registry/openshift/e2e/aws/aws-load-balancer/openshift-e2e-aws-aws-load-balancer-commands.sh:7]: I've been trying to find out if the repository is actually available when this script is run and I'm not sure if it is. If it's not then refactor it to use the same environment variables for the namespace as the one used by the operator-subscribe.

Comment (thejasn) [reply ci-operator/step-registry/openshift/e2e/aws/aws-load-balancer/openshift-e2e-aws-aws-load-balancer-commands.sh:7]: Couldn't find any docs regarding this. Couldn't find any other steps doing this either. According to the CI docs, the repo is cloned as layers ([as src](https://docs.ci.openshift.org/docs/architecture/ci-operator/#building-artifacts)) and doesn't really mention if steps (pre and post) are executed in the same layers (doesn't allow adding `from: src` tag either).

Comment (thejasn) [reply ci-operator/step-registry/openshift/e2e/aws/aws-load-balancer/openshift-e2e-aws-aws-load-balancer-commands.sh:7]: >  refactor it to use the same environment variables for the namespace as the one used by the operator-subscribe.
We can't reuse `OO_INSTALL_NAMESPACE` since the namespace won't exist during this step (will be created later on). New variables can be introduced, though, if needed.

File: ci-operator/config/openshift/aws-load-balancer-operator/openshift-aws-load-balancer-operator-main.yaml (hunk 4 above)
Comment (thejasn) [reply ci-operator/config/openshift/aws-load-balancer-operator/openshift-aws-load-balancer-operator-main.yaml:35]: Yes, for some reason, the CI fails in the bundle-gen step without promotion. (not entirely sure why)

File: ci-operator/config/openshift/aws-load-balancer-operator/openshift-aws-load-balancer-operator-main.yaml (hunk 6 above)
Comment (thejasn) [reply ci-operator/config/openshift/aws-load-balancer-operator/openshift-aws-load-balancer-operator-main.yaml:69]: Workflows seem to be more of a reusable component, since the additional step we added is too specific, I'm inclined to leave it as is for now.

File: ci-operator/step-registry/openshift/e2e/aws/aws-load-balancer/openshift-e2e-aws-aws-load-balancer-commands.sh (hunk 1 above)
Comment (thejasn) [reply ci-operator/step-registry/openshift/e2e/aws/aws-load-balancer/openshift-e2e-aws-aws-load-balancer-commands.sh:7]: We can reuse `OO_PACKAGE` and `OO_INSTALL_NAMESPACE` for the secret ref and the name of the `credentialsrequest`. Is that what you meant?

Comment (arjunrn) [reply ci-operator/step-registry/openshift/e2e/aws/aws-load-balancer/openshift-e2e-aws-aws-load-balancer-commands.sh:7]: This [step](https://github.com/openshift/release/blob/10549abd27aa8738f17d56a75576be4329e54f22/ci-operator/step-registry/baremetalds/metallb-e2e/test/baremetalds-metallb-e2e-test-ref.yaml#L3) uses `from: src`. 
> We can't reuse OO_INSTALL_NAMESPACE since the namespace won't exist during this step (will be created later on). New variables can be introduced, though, if needed.
Yes, parameterize the targets of CredentialsRequest.

File: ci-operator/config/openshift/aws-load-balancer-operator/openshift-aws-load-balancer-operator-main.yaml [hunk 7]
```diff
@@ -49,6 +64,31 @@ tests:
...
+    dependencies:
+      OO_INDEX: ci-index-aws-load-balancer-operator-bundle
+    env:
+      OO_CHANNEL: alpha
+      OO_INSTALL_NAMESPACE: aws-load-balancer-operator
+      OO_PACKAGE: aws-load-balancer-operator
+      OO_TARGET_NAMESPACES: '!install'
+    post:
+    - chain: ipi-aws-post
```
[the remark this answers is not in the document]
Comment (thejasn) [reply ci-operator/config/openshift/aws-load-balancer-operator/openshift-aws-load-balancer-operator-main.yaml:78]: These steps are explicitly given, since we are using a modified version of an existing workflow ([optional-operators-ci-aws](https://github.com/openshift/release/blob/master/ci-operator/step-registry/optional-operators/ci/aws/optional-operators-ci-aws-workflow.yaml)).

File: ci-operator/config/openshift/aws-load-balancer-operator/openshift-aws-load-balancer-operator-main.yaml [hunk 8]
```diff
@@ -49,6 +64,31 @@ tests:
...
+    pre:
+    - chain: ipi-aws-pre
+    - ref: openshift-e2e-aws-aws-load-balancer
+    - ref: optional-operators-subscribe
```
[the remark this answers is not in the document]
Comment (thejasn) [reply ci-operator/config/openshift/aws-load-balancer-operator/openshift-aws-load-balancer-operator-main.yaml:82]: We are just injecting our step in between the existing steps present in the ([optional-operators-ci-aws](https://github.com/openshift/release/blob/master/ci-operator/step-registry/optional-operators/ci/aws/optional-operators-ci-aws-workflow.yaml)) workflow. 

Ref slack thread: https://coreos.slack.com/archives/CBN38N3MW/p1648630615047199

File: ci-operator/config/openshift/aws-load-balancer-operator/openshift-aws-load-balancer-operator-main.yaml (hunk 7 above)
[the remark this answers is not in the document]
Comment (alebedev87) [reply ci-operator/config/openshift/aws-load-balancer-operator/openshift-aws-load-balancer-operator-main.yaml:78]: I see, then I agree with [Arjun's point](https://github.com/openshift/release/pull/27205#discussion_r838617772) on the reason of why we need to specify the workflow.

File: core-services/image-mirroring/openshift/mapping_origin_4_11 [hunk 9]
```diff
@@ -18,6 +18,7 @@ registry.ci.openshift.org/origin/4.11:aws-efs-csi-driver quay.io/openshift/origi
 registry.ci.openshift.org/origin/4.11:aws-efs-csi-driver-operator quay.io/openshift/origin-aws-efs-csi-driver-operator:4.11 quay.io/openshift/origin-aws-efs-csi-driver-operator:4.11.0 quay.io/openshift/origin-aws-efs-csi-driver-operator:latest
 registry.ci.openshift.org/origin/4.11:aws-efs-csi-operator-create-efs quay.io/openshift/origin-aws-efs-csi-operator-create-efs:4.11 quay.io/openshift/origin-aws-efs-csi-operator-create-efs:4.11.0 quay.io/openshift/origin-aws-efs-csi-operator-create-efs:latest
 registry.ci.openshift.org/origin/4.11:aws-efs-utils-base quay.io/openshift/origin-aws-efs-utils-base:4.11 quay.io/openshift/origin-aws-efs-utils-base:4.11.0 quay.io/openshift/origin-aws-efs-utils-base:latest
+registry.ci.openshift.org/origin/4.11:aws-load-balancer-operator quay.io/openshift/origin-aws-load-balancer-operator:4.11 quay.io/openshift/origin-aws-load-balancer-operator:4.11.0 quay.io/openshift/origin-aws-load-balancer-operator:latest
```
[the remark this answers is not in the document]
Comment (thejasn) [reply core-services/image-mirroring/openshift/mapping_origin_4_11]: Fixed it.

File: ci-operator/config/openshift/aws-load-balancer-operator/openshift-aws-load-balancer-operator-main.yaml (hunk 4 above)
Comment (thejasn) [reply ci-operator/config/openshift/aws-load-balancer-operator/openshift-aws-load-balancer-operator-main.yaml]: Nvm, turns out it wasn't required.

File: ci-operator/step-registry/aws-load-balancer/pre-install/aws-load-balancer-pre-install-commands.sh [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+set -o pipefail
+
+curl -L https://github.com/mikefarah/yq/releases/download/3.3.0/yq_linux_amd64 -o /tmp/yq && chmod +x /tmp/yq
+
+AWS_LOAD_BALANCER_OPERATOR_SRC_DIR="/go/src/github.com/openshift/aws-load-balancer-operator"
+AWS_CREDENTIALS_REQUEST="/tmp/operator-credentials-request.yaml"
+
+touch "${AWS_CREDENTIALS_REQUEST}"
+
+cat ${AWS_LOAD_BALANCER_OPERATOR_SRC_DIR}/hack/operator-credentials-request.yaml > "${AWS_CREDENTIALS_REQUEST}"
```
Comment (alebedev87) [inline ci-operator/step-registry/aws-load-balancer/pre-install/aws-load-balancer-pre-install-commands.sh:14]:
```suggestion
cp ${AWS_LOAD_BALANCER_OPERATOR_SRC_DIR}/hack/operator-credentials-request.yaml "${AWS_CREDENTIALS_REQUEST}"
```

**Review by alebedev87 — COMMENTED** [review]
LGTM

**Review by petr-muller — APPROVED** [review]
